### DISCUSSION TITLE
Maybe have solution for google login

### DISCUSSION THREAD

**Discussion by jarno9981** [cefsharp/CefSharp#3538] [General]
http://www.chromium.org/developers/how-tos/api-keys

I readed a artical and they said this is the solution to the login problems

**Comment by amaitland** [comment]
I'm not aware of a example, try asking on stackoverflow.

**Reply by jarno9981** [reply]
https://superuser.com/questions/1100270/i-cant-log-in-chromium-i-was-a-chrome-user-before-this-so-should-make-another

This the topic i got and says something about api to sync google things

**Comment by timaiv** [comment]
My company (on win i implemented this) migrated from cef based (that also has problem on ios and android chromium based browsers) to "in browser" based oauth google authentification 1.5 years ago. (from [Google Samples)](https://github.com/googlesamples/oauth-apps-for-windows/blob/master/OAuthDesktopApp/OAuthDesktopApp/MainWindow.xaml.cs)
Works fine. 
Only one problem-we cant know when browser\tab\process is killed, to refresh our app UI.
___
U can try change user-agent (saw in forum to this problem) to `firefox`, last time it was working.

**Reply by jarno9981** [reply]
@timaiv  i try with other agent like the webview 2 control but problem is you cant deploy the webview 2 it alwalys fails   ( i like cefsharp more than webview 2)

**Reply by timaiv** [reply]
Misunderstood here.
I mean change "user agent" header in every http request. (keep using cefsharp). It was working for me, but it bad solution.

The User-Agent request header is a characteristic string that lets servers and network peers identify the application, operating system, vendor, and/or version of the requesting user agent.
